pyClient/README.md

```diff
@@ -45,11 +55,192 @@ and fails to fetch it (or fails to fetch the right version).
+
+# Compile and deploy
+(env)$ zeth deploy
+
+# Share the instance file with all clients
+$ cp zeth-instance.json <destination>
+```
+
+## User setup
+
+To set up her client, Alice must setup all client files mentioned above:
+```console
+# Create a clean client directory
+$ mkdir alice
+$ cd alice
+
+# Specify an eth-address file for an (unlocked) Ethereum account
+$ echo 0x.... > eth-address
+
+# Copy the instance file (received from the deployer)
+$ cp <shared-instance-file> zeth-instance.json
+
+# Generate new secret (zeth-key.json)and public key (zeth-key.json.pub)
+$ zeth key
+
+# Share the public address with other users
+$ cp zeth-key.json.pub <destination>
+```
+
+With these files in place, `zeth` commands invoked from inside this directory
+can perform actions on behalf of Alice.  We call this Alice's *client directory*
+below, and assume that all commands are executed in a directory with these
+files.
+
+## Depositing funds
+
+Alice can deposit funds from the address in `eth-address` into the Zeth
+instance, as a private Zeth note, spendable with her secret key `zeth-key.json`:
+
+```console
+# Deposit 10 ether, creating a Zeth note owned by Alice
+(env)$ zeth deposit 10
+```
+
+Note that for this deposit to be reflected in Alice's client direectory, she
+must sync with the blockchain as described below.
+
+## Receiving transactions
+
+The following command scans the blockchain for any new transactions addressed to
+the public key `zeth-key.json.pub`:
+
+```console
+# Check all new blocks for notes addressed to `zeth-key.json.pub`, storing them
+# in the ./notes directory.
+(env)$ zeth sync
+```
+
+Any notes found are stored in the `./notes` direectory, as files containing the
+secret data required to spend them.  This forms a very primitive Zeth "wallet".
+
+```console
+# List all notes received by this client
+$ zeth notes
+```
+lists information about all known notes belonging to the current user.
+
+## Spending and withdrawing notes
+
+`zeth send` and `zeth withdraw` require inputs in the form of Zeth note
+commitments, and outputs (for `zeth send`) in the form of Zeth addresses and
+amounts.
+
+**Inputs** are Zeth notes owned by the current client, which should be visible
+via `zeth notes`.  Either the integer "address" or the truncated commitment
+value (8 hex chars) can be used to specify which notes to use as inputs.
+
+**Outputs** are given as the Zeth public key and a value, separated by a comma
+`,`.  The form of the public key is exactly as in the `zeth-key.json.pub` file.
+That is, two 32 byte hex values separated by a colon `:`.
+
+Examples:
+```console
+$ zeth notes
+b1a2feaf: value=200, addr=0
+eafe5f84: value=100, addr=2
+
+$ zeth spend \
+    --in eafe5f84 \                       # "eafe5f84: value=100, addr=2"
+    --in 0 \                              # "b1a2feaf: value=200, addr=0"
+    --out d77f...0e00:cc7c....7f76,120 \  # 120 to this addr
+    --out 3a43...fd3b:9fc8....b838,180    # 180 to this addr
```

Comment:
Do we have a way to specify `v_in` to deposit funds?

I'd be in favor of merging all of these commands (`spend`, `withdraw` and so on) into a single command, like `zeth pay` (as in: https://github.com/clearmatics/zeth/issues/8#issuecomment-555618678), or `zeth mix` (to remain consistent with the function name on the mixer).

Separating `deposits`, `withdrawals` and so on, is the type of anti-patterns that we should avoid at any cost (at all stages of the project) as it leaks a lot of *exact* data to an adversary. Sticking to a single functionality would enable users to deposit, withdraw, transfer notes into a single command which provides more flexibility and which is better as it adds noise (the adversary does not exactly know what the user is doing, ie: the user may be able to withdraw and transfer in the same transaction, and so on).

To do so, we'd need, something like:
`zeth mix --in <note1> --in <note2> --vin <v_in> --vout <v_out> --out <a_pk1> --out <a_pk2>`
(we can use default values if the flags are not set)

Reply:
Replaced `deposit`, `withdraw` and `send` with a single `mix` function.  User must compute input and output values correectly, or the command throws an error.